main-command/src/main/scala/sbt/internal/client/NetworkClient.scala

```diff
@@ -9,94 +9,242 @@ package sbt
 package internal
 package client
 
-import java.io.{ File, IOException }
+import java.io.{ File, IOException, InputStream, PrintStream }
+import java.lang.ProcessBuilder.Redirect
+import java.net.Socket
+import java.nio.channels.ClosedChannelException
+import java.nio.file.Files
 import java.util.UUID
 import java.util.concurrent.atomic.{ AtomicBoolean, AtomicReference }
+import java.util.concurrent.{ ConcurrentHashMap, LinkedBlockingQueue, TimeUnit }
 
+import sbt.internal.client.NetworkClient.Arguments
 import sbt.internal.langserver.{ LogMessageParams, MessageType, PublishDiagnosticsParams }
 import sbt.internal.protocol._
-import sbt.internal.util.{ ConsoleAppender, LineReader }
+import sbt.internal.util.{ ConsoleAppender, ConsoleOut, Signals, Terminal, Util }
 import sbt.io.IO
 import sbt.io.syntax._
 import sbt.protocol._
 import sbt.util.Level
+import sjsonnew.BasicJsonProtocol._
+import sjsonnew.shaded.scalajson.ast.unsafe.{ JObject, JValue }
 import sjsonnew.support.scalajson.unsafe.Converter
 
-import scala.collection.mutable.ListBuffer
-import scala.sys.process.{ BasicIO, Process, ProcessLogger }
+import scala.annotation.tailrec
+import scala.collection.mutable
+import scala.concurrent.duration._
 import scala.util.control.NonFatal
-import scala.util.{ Failure, Success }
+import scala.util.{ Failure, Properties, Success }
+import Serialization.{
+  CancelAll,
+  attach,
+  cancelRequest,
+  promptChannel,
+  systemIn,
+  systemOut,
+  terminalCapabilities,
+  terminalCapabilitiesResponse,
+  terminalPropertiesQuery,
+  terminalPropertiesResponse
+}
+import NetworkClient.Arguments
+
+trait ConsoleInterface {
+  def appendLog(level: Level.Value, message: => String): Unit
+  def success(msg: String): Unit
+}
 
-class NetworkClient(configuration: xsbti.AppConfiguration, arguments: List[String]) { self =>
-  private val channelName = new AtomicReference("_")
+class NetworkClient(
+    console: ConsoleInterface,
+    arguments: Arguments,
+    inputStream: InputStream,
+    errorStream: PrintStream,
+    printStream: PrintStream,
```

Comment:
Could you add Scaladoc on the significance of passing these streams into `NetworkClient` while your knowledge is fresh in your mind please?

Reply:
https://github.com/sbt/sbt/pull/5620/commits/d353e012aa4a311ba9c8f7357551d6178419ddd7